KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/RegexQuery.java

```diff
@@ -101,6 +101,15 @@
     private String field = Server.Schema.CONTENT_STR.toString();
 
     /**
+     * The following map is an optimization to ensure that we reuse
+     * the same keyword hit String object across all hits. Even though we 
+     * benefit from G1GC String deduplication, the overhead associated with
+     * creating a new String object for every KeywordHit can be significant 
+     * when the number of hits gets large.
+     */
+    private final HashMap<String, String> keywordsFoundAcrossAllDocuments;
```

Comment:
I assume this was explored, but I will just mention it incase:  Was string interning looked at as an alternative to this map.